### PR TITLE
Phase2-hgx319A Test RecHitTool for partial wafers of HGCal and also add a few utility methods for wafer

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Test RecHitTool for partial wafers of HGCal and also add a few utility methods for wafer

#### PR validation:

Use test workflows in RecoLocalCalo/HGCalRecAlgos/test

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

**Comment by bsunanda** [comment]
@AdrianoDee, @srimanob, @clacaputo, @jpata Could you review and if possible approve this PR?

File: SimG4CMS/Calo/src/HGCalNumberingScheme.cc [hunk 1]
```diff
@@ -51,7 +51,7 @@ uint32_t HGCalNumberingScheme::getUnitID(int layer, int module, int cell, int iz
       cellV = HGCalTypes::getUnpackedCellV(cell);
     } else if (mode_ != HGCalGeometryMode::Hexagon8) {
       double xx = (pos.z() > 0) ? pos.x() : -pos.x();
-      hgcons_.waferFromPosition(xx, pos.y(), layer, waferU, waferV, cellU, cellV, waferType, wt, false, false);
+      hgcons_.waferFromPosition(xx, pos.y(), layer, waferU, waferV, cellU, cellV, waferType, wt, false, true);
```
Comment (srimanob) [inline SimG4CMS/Calo/src/HGCalNumberingScheme.cc:54]: No effect of changing this boolean? I don't see what about it in the PR description.

Comment (bsunanda) [reply SimG4CMS/Calo/src/HGCalNumberingScheme.cc]: This was a debug switch. I was debugging partial wafers. Now I have put it back to false

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@civanch,  @clacaputo, @AdrianoDee, @srimanob,  @jpata Could you sign this PR now?

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**